### DISCUSSION TITLE
[HowTo] Building hyprland on debian sid

### DISCUSSION THREAD

**Discussion by werdahias** [hyprwm/Hyprland#1797] [General]
Should work on a recent ubuntu too, untested though.

Prerequisites:

`sudo apt install build-essential git ninja-build meson &&  sudo apt-get build-dep wlroots`

Building:

`git clone --recursive https://github.com/hyprwm/Hyprland`

If you want to use a specific release use 
`git clone --recursive https://github.com/hyprwm/Hyprland -b VERSION`

```
cd Hyprland
meson build
ninja -C build
sudo ninja -C build install
```
Then edit your hyprland.conf and you're good to go.

**Comment by ghost** [comment]
how to uninstall it

**Reply by werdahias** [reply]
sudo ninja -C build uninstall

**Comment by krlade** [comment]
> untested though

Have you tested it on debian sid? These are not all of the needed dependencies

**Reply by werdahias** [reply]
Sure did, but this is for whatever hyprland version was out in march. The current one needs additional deps, you're right. Since debian is also in freeze atm you can't build the current hyprland since it needs bewer libraries not present yet

The untested bit was referring to ubuntu

**Comment by Joelangelo** [comment]
Has anyone come up with a way to install Hyprland that is up to date with Debian 12?
My attempt is currently failing at the "meson build" step because i have version 1.21 och wayland-server instead of ">= 1.22".
Not sure if im lacking any more dependencies though since i cant get any further for now.

**Comment by manatails** [comment]
Hyprland v0.24.1 compiles fine on Debian Sid as of the time of writing.
I had to clone optional submodules otherwise the build would fail.

```
git clone --recursive https://github.com/hyprwm/Hyprland -b v0.24.1
cd Hyprland
cd subprojects
git clone https://gitlab.freedesktop.org/emersion/libdisplay-info
git clone https://github.com/emersion/libliftoff
cd ..
meson build
ninja -C build
sudo ninja -C build install
```

**Reply by HystericalHatter** [reply]
0.27.0 seems to compile fine too following the same steps (I didn't need to clone libdisplay-info though).
I did it in a lxc container in which I installed many packages trying to get the compilation right, so I think it needs to be confirmed in a clean environment, but so far it seems good.
Thanks !

Edit : I forgot to mention that I had to install wayland-server from experimental since Hyprland requires >=1.22 and only 1.21 is available in sid ATM.

**Reply by werdahias** [reply]
libliftoff was updated in sid so both submodules shouldn't be needed

**Comment by Skovrup1** [comment]
Got latest version commit ec26962 to compile on sid using 
Prerequisites from werdahias:
`sudo apt install build-essential git ninja-build meson  &&  sudo apt-get build-dep wlroots`
Also had to add these packages for cairo and pango:
`sudo apt install libcairo2-dev libpango1.0-dev`

And by following manatails steps

**Comment by enigma9o7** [comment]
For what its worth, I was also able to build current git version (28.whatever) on bookworm (debian 12 stable) following manatails directions too, after doing simplebackport from trixie on 3 packages: wayland, wayland-protocols, and libinput.

Before doing the backports, meson told me wayland, then wayland-protocols were too old.  But libinput just gave errors during build, it didn't warn me that wasn't new enough.  So I think that's a minor bug, meson should tell user need newer libinput.

**Comment by MakiseKurisu** [comment]
I don't want to deal with back-porting Wayland, so I use v0.24.1, which is the last version with wlroots targeting Wayland 1.21. I only need Hyprland as a Looking Glass launcher, so I don't mind the old version.

**Comment by gitmanbloggus** [comment]
Should work on the raspberry pi because it uses debian 12 and most packages are available, haven't tried though

**Comment by eriad25** [comment]
I think you can build it on 12 and run it but problem is latest versions require newer cmake which isn't avaliable on 12 so maybe using repos from trixie to update software is good idea but not the system yet. I've seen people making hyprland work on debian 12 without problem. Cmake on debian 12 is 3.5.2 i think and as of now it requires 3.6 or newer which isn't avaliable for debian 12 and you either have to install them from website or swap to either trixie or backports to get newer stuff.

**Comment by PerkelSimon** [comment]
hi, 

IM on trixie, not testing, I mean I removed testing in sources.list to trixie, so if I want to install the latest hyprland (0.47.2) do you think it will work?
thanks

**Reply by gitmanbloggus** [reply]
> hi,
> 
> IM on trixie, not testing, I mean I removed testing in sources.list to trixie, so if I want to install the latest hyprland (0.47.2) do you think it will work? thanks

Trixie is testing
it should work
